File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        f = {}

        for p in nums:
            if p not in f:
                f[p] = 1
            else:
                f[p] = f[p] + 1

        nums = sorted(f.items(), key=lambda x: x[1], reverse=True)

        r = []

        for p in range(k):
            r.append(nums[p][0])

        return r

        

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) 
    {
        HashMap<Integer,Integer> map=new HashMap<>();
        List<Integer> unique=new ArrayList<>();
        for(int i=0;i<nums.length;i++)
        {
            if(!map.containsKey(nums[i]))
            {
                unique.add(nums[i]);
            }
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }
        unique.sort((a,b)->map.get(b)-map.get(a));
        int[] ans=new int[k];
        for(int i=0;i<k;i++)
        {
            ans[i]=unique.get(i);
        }
        return ans;
    }
}